assets/js/base/context/editor/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/contexts').EditorDataContext} EditorDataContext
+ */
+
+const EditorContext = createContext( {
+	isEditor: false,
+	currentPostId: 0,
+} );
+
+/**
+ * @return {EditorDataContext} Returns the editor data context value
+ */
+export const useEditorContext = () => {
+	return useContext( EditorContext );
+};
+
+/**
+ * Editor provider
+ *
+ * @param {Object}  props              Incoming props for the provider.
+ * @param {*}       props.children     The children being wrapped.
+ * @param {number}  props.currentPostId The post being edited.
+ */
+export const EditorProvider = ( { children, currentPostId = 0 } ) => {
```

Comment:
Is currrentPostId something that should be derived internally in the provider or does it need explicitly passed in?  What about keeping this, but if it's 0, use the `core/editor` selector to get it?

Reply:
I did it this way so you could control the post ID given to a current context. But yes, we can do it again inside if thats allowed :)